services/debugger/cache_test.go

```diff
@@ -0,0 +1,69 @@
+package debugger
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"time"
+)
+
+var _ = Describe("cache", func() {
+
+	Context("cache testing", func() {
+		testKey := "test_key"
+		testValue := []byte("test_value")
+
+		It("Cache init", func() {
+			var c CacheT
+			c.init()
+			Expect(c.size).NotTo(Equal(0))
+			Expect(c.keyTTL).NotTo(Equal(0))
+			Expect(c.cacheMap).NotTo(BeNil())
+		})
+
+		It("Cache update", func() {
+			var c CacheT
+			c.Update(testKey, testValue)
+			Expect(len(c.cacheMap)).To(Equal(1))
+			Expect(len(c.cacheMap[testKey].data)).To(Equal(1))
+			Expect(c.cacheMap[testKey].data[0]).To(Equal(testValue))
+		})
+
+		It("Cache timeout", func() {
+			var c CacheT
+			c.init()
+			c.keyTTL = time.Second
+			c.cleanupFreq = time.Second
+			Expect(c.keyTTL).To(Equal(time.Second))
+			c.Update(testKey, testValue)
+			Expect(len(c.cacheMap)).To(Equal(1))
+			Expect(len(c.cacheMap[testKey].data)).To(Equal(1))
+			Expect(c.cacheMap[testKey].data[0]).To(Equal(testValue))
+			time.Sleep(time.Second * 2)
+			Expect(len(c.cacheMap)).To(Equal(0))
```

Comment:
Avoid sleeps in tests, eventually checks for a condition in a loop and is faster and more robust.
```suggestion
			Eventually(func() int {
			return len(c.cacheMap) }).To(Equal(0))
```

Reply:
done